weaviate/connect/connection.py

```diff
@@ -458,6 +470,36 @@ def timeout_config(self, timeout_config: Union[Tuple[Real, Real], Real]):
     def proxies(self) -> dict:
         return self._proxies
 
+    def wait_for_weaviate(self, wait_for_weaviate: Optional[int]):
+        """
+        Waits until weaviate is ready or the timelimit given in 'wait_for_weaviate' has passed.
+
+        Parameters
+        ----------
+        wait_for_weaviate : Optional[int]
+            Describes how long the client will wait for weaviate to start in seconds.
+
+        Raises
+        ------
+        ConnectionError
+            If weaviate takes longer than the timelimit to respond.
+        """
+
+        ready_url = self.url + self._api_version_path + "/.well-known/ready"
+        for _i in range(0, wait_for_weaviate, 1):
+            try:
+                status_code = requests.get(
+                    ready_url,
+                ).status_code
+                if status_code != 200:
```

Comment:
Something that I would like to change for all status checks: Do not check for specific value but rather check if not an error status code. All `2XX` values are successful ones, all `4XX` are error ones and all `5XX` server errors,  and the `3XX` are redirects. More on this [here](https://www.oncrawl.com/oncrawl-seo-thoughts/meaning-of-the-different-https-status-code/#:~:text=300s%3A%20redirection%3A%20the%20request%20has,destination%20page%20is%20not%20correct.). So lets change it so that if status code is `>= 400` we retry.

FYI: There is a requests method that does this: https://requests.readthedocs.io/en/latest/_modules/requests/models/#Response.raise_for_status